File: lib_common/src/main/java/com/abctime/lib_common/data/network/BaseApiService.java
package com.abctime.lib_common.data.network;

import com.google.gson.JsonObject;

import java.util.Map;

import io.reactivex.Flowable;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.PartMap;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;
import retrofit2.http.Streaming;
import retrofit2.http.Url;

/**
 * Created by xmren on 2018/3/9.
 */

public interface BaseApiService {
    @GET("{url}")
    Flowable<JsonObject> executeGet(
            @Path("url") String url,
            @QueryMap Map<String, String> maps
    );


    @POST("{url}")
    Flowable<JsonObject> executePost(
            @Path("url") String url,
            //  @Header("") String authorization,
            @QueryMap Map<String, String> maps);

    @POST("{url}")
    Flowable<JsonObject> json(
            @Path("url") String url,
            @Body RequestBody jsonStr);

    @Multipart
    @POST("{url}")
    Flowable<JsonObject> upLoadFile(
            @Path("url") JsonObject url,
            @Part("image\"; filename=\"image.jpg") RequestBody requestBody);

    @POST("{url}")
    Flowable<JsonObject> uploadFiles(
            @Path("url") String url,
            @Path("headers") Map<String, String> headers,
            @Part("filename") String description,
            @PartMap() Map<String, RequestBody> maps);

    @Streaming
    @GET
    Flowable<ResponseBody> downloadFile(@Url String fileUrl);
}

File: lib_common/src/main/java/com/wujia/lib_common/data/network/BaseApiService.kt
package com.wujia.lib_common.data.network

import com.google.gson.JsonObject

import io.reactivex.Flowable
import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.Part
import retrofit2.http.PartMap
import retrofit2.http.Path
import retrofit2.http.QueryMap
import retrofit2.http.Streaming
import retrofit2.http.Url

/**
 * Created by xmren on 2018/3/9.
 */

interface BaseApiService {


    @Multipart
    @POST("{url}")
    fun upLoadFile(
            @Path("url") url: JsonObject,
            @Part("image\"; filename=\"image.jpg") requestBody: RequestBody): Flowable<JsonObject>

    @Streaming
    @GET
    fun downloadFile(@Url fileUrl: String): Flowable<ResponseBody>
}
